src/lib.rs

```diff
@@ -135,49 +135,50 @@
 #![forbid(unsafe_code)]
 #![warn(rust_2018_idioms, unreachable_pub)]
 
-use smallvec::SmallVec;
 use std::str::from_utf8;
 
+use smallvec::SmallVec;
+
 mod node;
 mod parser;
 
 pub use node::{Node, NodeKind};
 pub use parser::{Kind, Parser, Piece, Position};
 
 #[derive(Debug)]
-pub struct PathTree<'a, T> {
+pub struct PathTree<T> {
     id: usize,
-    routes: Vec<(T, Vec<Piece<'a>>)>,
-    pub node: Node<'a, usize>,
+    routes: Vec<(T, Vec<Piece>)>,
+    pub node: Node<usize>,
 }
 
-impl<'a, T> Default for PathTree<'a, T> {
+impl<T> Default for PathTree<T> {
     fn default() -> Self {
         Self::new()
     }
 }
 
-impl<'a, T> PathTree<'a, T> {
+impl<T> PathTree<T> {
     pub fn new() -> Self {
         Self {
             id: 0,
             routes: Vec::new(),
-            node: Node::new(NodeKind::String("".as_bytes()), None),
+            node: Node::new(NodeKind::String("".as_bytes().to_vec()), None),
         }
     }
 
-    pub fn insert(&mut self, path: &'a str, value: T) -> &mut Self {
+    pub fn insert(&mut self, path: &str, value: T) -> &mut Self {
```

Comment:
I think these lifetimes would be fine to keep. For PyO3 we need to copy the data into Python objects to return it anyway, so lifetimes on methods are fine. The problematic lifetimes are on the PathTree. Also I imagine most use cases don’t care too much about insertion times (ideally that would happen once at app startup) so I would _heavily_ bias toward optimizing the performance of find() over all else.